wgpu-core/src/device/mod.rs

```diff
@@ -2987,41 +2940,94 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
 
 #[derive(Clone, Debug, Error)]
 pub enum CreateBufferError {
-    #[error("`MAP` usage can only be combined with the opposite `COPY`, requested {0:?}")]
-    UsageMismatch(wgt::BufferUsage),
+    #[error(transparent)]
+    BindError(#[from] hal::device::BindError),
```

Comment:
We don't really want to propagate HAL errors up. They are *not* expected.
The idea of `wgpu` is that it's a safe abstraction over HAL, it has logic to ensure that HAL is used safely, and all errors except for DeviceLost and OutOfMemory are unexpected and should just be unwrapped internally.